tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -419,5 +419,42 @@ public void Set(TCarrier carrier, string key, string value)
                 return Enumerable.Empty<string?>();
             }
         }
+
+        internal void AddBaggageToSpanAsTags(ISpan span, Baggage? baggage)
+        {
+            if (baggage is null or { Count: 0 })
+            {
+                return;
+            }
+
+            var settings = Tracer.Instance.Settings;
+            var baggageTagKeys = settings.BaggageTagKeys;
+
+            if (string.IsNullOrWhiteSpace(baggageTagKeys))
+            {
+                // feature disabled
+                return;
+            }
+
+            if (baggageTagKeys == "*")
+            {
+                // add all baggage items as tags
+                foreach (var item in baggage)
+                {
+                    span.SetTag(item.Key, item.Value);
+                }
+                return;
+            }
+
+            // add only specified baggage items as tags
+            var allowedKeys = baggageTagKeys.Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+            foreach (var key in allowedKeys)
+            {
+                if (baggage.TryGetValue(key, out var value))
+                {
+                    span.SetTag(key, value);
+                }
+            }
```

Comment:
`StringSplitOptions.TrimEntries` was added in .NET 5, so this code won't compile for earlier versions of .NET. Instead, you can remove `StringSplitOptions.TrimEntries` here and call `Trim()` manually below:

```suggestion
            var allowedKeys = baggageTagKeys.Split(',', StringSplitOptions.RemoveEmptyEntries);

            foreach (var key in allowedKeys)
            {
                var trimmedKey = key.Trim();

                if (baggage.TryGetValue(trimmedKey, out var value))
                {
                    span.SetTag(trimmedKey, value);
                }
            }
```